### PR TITLE
Fixed issue #232 - RFC: add support for computing the inverse/arc cosecant (acsc)

### PR DESCRIPTION
**Pull request by spneshaei**
## Description

> What is the purpose of this pull request?

This pull request:

-   Adds support for acsc mathematical function in the base library.
-   Adds related documentation for acsc function.
-   Adds examples of using acsc function.
-   Adds JS benchmarks for the acsc function.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   fixes issue #232

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-  Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by spneshaei** [comment]
This PR has commits considering the same notes and principles noted in https://github.com/stdlib-js/stdlib/pull/476#issuecomment-1026530207

File: lib/node_modules/@stdlib/math/base/special/acsc/docs/repl.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+{{alias}}( x )
+    Computes the arccosecant of a number.
+
+    If `|x| < 1`, the function returns `NaN`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acsc/docs/repl.txt:5]: Is this true for this function?

**Comment by kgryte** [comment]
Same general comments as in #476.

**Review by kgryte — APPROVED** [review]
This PR needs a fair amount of refactoring and clean-up; however, I'll do this on our end. Thanks for the contribution @spneshaei.

**Comment by kgryte** [comment]
Refactored and cleaned in https://github.com/stdlib-js/stdlib/commit/05af4dd44f3b1c81df6071fda15f75713c4fce69.

**Pull request merged**